cmake/kokkos_arch.cmake

```diff
@@ -724,6 +729,15 @@ FOREACH(ARCH IN LISTS SUPPORTED_AMD_ARCHS)
   CHECK_AMDGPU_ARCH(${ARCH} ${FLAG})
 ENDFOREACH()
 
+IF(KOKKOS_IMPL_AMDGPU_FLAGS)
+  IF (NOT AMDGPU_ARCH_ALREADY_SPECIFIED)
+    MESSAGE(FATAL_ERROR "When IMPL_AMDGPU_FLAGS is set the architecture autodectection is disabled. "
+      "Please explicitly set the GPU architecture.")
+  ENDIF()
+  GLOBAL_APPEND(KOKKOS_AMDGPU_OPTIONS "${KOKKOS_IMPL_AMDGPU_FLAGS}")
+  GLOBAL_APPEND(KOKKOS_LINK_OPTIONS "${KOKKOS_IMPL_AMDGPU_LINK}")
```

Comment:
Is that not going to set it as target property and export it?

Reply:
Yes, that's what I wanted to do.